lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3627,7 +3627,10 @@ bool SPIRVToLLVM::transDecoration(SPIRVValue *BV, Value *V) {
 
   transIntelFPGADecorations(BV, V);
   transMemAliasingINTELDecorations(BV, V);
-  transVarDecorationsToMetadata(BV, V);
+
+  // Decoration metadata is only enabled in SPIR-V friendly mode
+  if (BM->getDesiredBIsRepresentation() == BIsRepresentation::SPIRVFriendlyIR)
+    transVarDecorationsToMetadata(BV, V);
```

Comment:
This 'if' might be  removed in the future, making the Decoration metadata be emitted by default and be transformed to OpenCL representation in SPIRVToOCL pass, but lets first see new features being developed atop of this infrastructure.